### PR TITLE
Feature/timeline mapeditor merged

### PR DESCRIPTION
**Pull request by rubenwiersma**
Timeline is now functional. Saving functionality will come next week.
TimePoints are deleted and CameraActions are too.

### CODE REVIEW

File: src/main/webapp/app/scripting/scripting-new-dialog.html [hunk 1]
```diff
@@ -4,22 +4,14 @@
...
-            <label for="id">ID</label>
-            <input type="text" class="form-control" id="id" name="id"
-                    ng-model="vm.cameraAction.id" readonly />
-        </div>
         <div class="form-group">
-            <label class="control-label" for="field_name">Name</label>
-            <input type="text" class="form-control" name="name" id="field_name"
-                    ng-model="vm.cameraAction.name"
-                     />
+            <label for="cameraAction">Camera action</label>
```
Comment (hungnguyen94) [inline src/main/webapp/app/scripting/scripting-new-dialog.html]: This label still uses cameraAction.

File: src/main/webapp/app/scripting/scripting.controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+     * @constructor
+     */
+    function ScriptingController ($rootScope, $scope, $state, Map, Cue, AlertService) {
+        var vm = this;
+
+        vm.cues = Cue.query();
+        getMapEntities();
+
+        function getMapEntities() {
+            Map.getDTO({id: 1}, function (result) {
```
Comment (hungnguyen94) [inline src/main/webapp/app/scripting/scripting.controller.js]: This should actually be changed to the mapId of the current Map in the project. That could be changed in the future though.

File: src/main/webapp/app/scripting/scripting-new-dialog.html (hunk 1 above)
Comment (rubenwiersma) [inline src/main/webapp/app/scripting/scripting-new-dialog.html]: Good catch! The label is for the text field, though. This text field is still called Camera Action, so the naming is logical this way.

File: src/main/webapp/app/scripting/timeline.directive.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+            /**
+             * Parses an int and returns the correct string for date parsing.
+             * @param year the year to parse
+             * @returns a string with a year
+             */
+            function parseIntAsYear(year) {
+                var current = "";
+
+                for (var j = 0; j < 4 - year.toString().length; ++j)
```
Comment (hungnguyen94) [inline src/main/webapp/app/scripting/timeline.directive.js]: Always use brackets in `for` constructs.

**Comment by hungnguyen94** [comment]
Everything looks good. I pushed a fix for the failing tests.

**Comment by hungnguyen94** [comment]
0.08%? :sob:

**Comment by hungnguyen94** [comment]
Alright, this can finally be merged!

**Pull request merged**